GWXtreme/parametrized_eos_sampler.py

```diff
@@ -179,7 +179,61 @@ def save_data(self):
         f.create_dataset('logp',data=np.array(self.logp))
         f.close()
         
+    def parse_samples(self,burn_in_frac = 0.5, thinning = None):
+        '''
+        This methods loads a .h5 file with the MCMC samples 
+        of EoS hyper-parameters and parses them for use by the 
+        plotting functions.
+        see https://emcee.readthedocs.io/en/stable/tutorials/autocorr/
+        for some documentation on choosing thinning and burn-in
+        
+        burn_in_frac  :: fraction of samples to discard from each chain
+                         for MCMC burn in. Default corresponds to discarding 
+                         half the samples in each chain.
+        
+        thinning      :: Number of samples to skip in each chain post
+                         burn in. "None" implements the default value 
+                         which is either (length of chain)/50 or half of
+                         the maximum integrated autocorrelation time. The 
+                         former is used in case autocorrelation analysis 
+                         yields throws non-convergence error. For no thinning 
+                         set thinning=1
+        
+        '''
+        
+        Ns=self.samples.shape
+        burn_in=int(Ns[0]*burn_in_frac)
+        samples=[]
+        
+        if thinning is None:
+            thinning=int(Ns[0]/50.)
```

Comment:
So, thinning is done by default. Why the choice of keeping every 50th sample?

Reply:
It's not entirely by default. Two things can happen:
1) user can supply it
2)if the user does not supply it, it will try to set the thinning as half of the maximum integrated autocorrelation time ([line 212](https://github.com/AnaryaRay1/GWXtreme/blob/2dcf2303d0a031898412b91f014812551efcd1c6/GWXtreme/parametrized_eos_sampler.py#L212)). Now in some cases, while calculating autocorrelation time, emcee will throw a convergence error if half of the autocorrelation time is larger than length of chain/50. In such cases, a default value of 50 is used. Details of emcee's autocorrelation analysis can be found [here](https://emcee.readthedocs.io/en/stable/tutorials/autocorr/). And the thinning = (1/2)*autocorrelation time is taken from various examples in emcee's documentation, such as [this one](https://emcee.readthedocs.io/en/stable/tutorials/monitor/)

Heuristically, these choices have been found to work in various cases Including the various set of mcmc runs done for the paper.

Reply:
That sounds good.